### ISSUE TITLE
An unexpected error has occurred.

### ISSUE THREAD

**Issue by lavi937** [conda/conda#5688]
Current conda install:

               platform : win-64
          conda version : 4.3.21
       conda is private : False
      conda-env version : 4.3.21
    conda-build version : not installed
         python version : 3.6.1.final.0
       requests version : 2.14.2
       root environment : C:\Users\User\Anaconda3  (writable)
    default environment : C:\Users\User\Anaconda3
       envs directories : C:\Users\User\Anaconda3\envs
                          C:\Users\User\AppData\Local\conda\conda\envs
                          C:\Users\User\.conda\envs
          package cache : C:\Users\User\Anaconda3\pkgs
                          C:\Users\User\AppData\Local\conda\conda\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : None
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.21 requests/2.14.2 CPython/3.6.1 Windows/10 Windows/10.0.14393
          administrator : False

`$ C:\Users\User\Anaconda3\Scripts\conda-script.py ..checkenv cmd.exe C:\Users\User\Anaconda3`

    Traceback (most recent call last):
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\cli\main.py", line 167, in main
        import conda.cli.activate as activate
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\cli\activate.py", line 12, in <module>
        from ..utils import shells
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\utils.py", line 13, in <module>
        from .gateways.disk.read import compute_md5sum
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\gateways\disk\read.py", line 22, in <module>
        from ...models.channel import Channel
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\models\channel.py", line 9, in <module>
        from ..base.context import context
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\base\context.py", line 21, in <module>
        from ..common.configuration import (Configuration, LoadError, MapParameter, PrimitiveParameter,
      File "C:\Users\User\Anaconda3\lib\site-packages\conda\common\configuration.py", line 42, in <module>
        from cytoolz.dicttoolz import merge
      File "C:\Users\User\Anaconda3\lib\site-packages\cytoolz\__init__.py", line 18, in <module>
        comp = compose
    NameError: name 'compose' is not defined

Getting this error while opening the anaconda command prompt.

Team, please help.

**Comment by ciacicode** [comment]
Was just installing a new env when I got a very similar error:

```
Current conda install:

               platform : linux-64
          conda version : 4.2.7
       conda is private : False
      conda-env version : 4.2.7
    conda-build version : not installed
         python version : 2.7.13.final.0
       requests version : 2.13.0
       root environment : /home/maria/miniconda2  (writable)
    default environment : /home/maria/miniconda2
       envs directories : /home/maria/miniconda2/envs
          package cache : /home/maria/miniconda2/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/linux-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
            config file : None
           offline mode : False


```

`$ /home/maria/miniconda2/bin/conda create -n khalenv python=2.7 anaconda`


```
Traceback (most recent call last):
      File "/home/maria/.local/lib/python2.7/site-packages/conda/exceptions.py", line 472, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/main.py", line 144, in _main
        exit_code = args.func(args, p)
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/main_create.py", line 68, in execute
        install(args, parser, 'create')
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/install.py", line 420, in install
        raise CondaRuntimeError('RuntimeError: %s' % e)
    CondaRuntimeError: Runtime error: RuntimeError: Invalid mode: <conda.install.FileMode object at 0x7fba0b2aff50>
```

**Comment by tyler-thetyrant** [comment]
@ciacicode Can you try with the most recent version of conda? We are currently at 4.3.22

**Comment by ciacicode** [comment]
2 hours ago I did 
`conda update conda`

and got this:

```
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /home/maria/miniconda2:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.3.22               |           py27_0         493 KB

The following packages will be UPDATED:

    conda: 4.3.11-py27_0 --> 4.3.22-py27_0

Proceed ([y]/n)? y

Fetching packages ...
conda-4.3.22-p 100% |################################| Time: 0:00:00   3.29 MB/s
Extracting packages ...
[      COMPLETE      ]|###################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : linux-64
          conda version : 4.2.7
       conda is private : False
      conda-env version : 4.2.7
    conda-build version : not installed
         python version : 2.7.13.final.0
       requests version : 2.13.0
       root environment : /home/maria/miniconda2  (writable)
    default environment : /home/maria/miniconda2
       envs directories : /home/maria/miniconda2/envs
          package cache : /home/maria/miniconda2/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/linux-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
            config file : None
           offline mode : False
```

`$ /home/maria/miniconda2/bin/conda update conda`


```
    Traceback (most recent call last):
      File "/home/maria/.local/lib/python2.7/site-packages/conda/exceptions.py", line 472, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/main.py", line 144, in _main
        exit_code = args.func(args, p)
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/main_update.py", line 65, in execute
        install(args, parser, 'update')
      File "/home/maria/.local/lib/python2.7/site-packages/conda/cli/install.py", line 420, in install
        raise CondaRuntimeError('RuntimeError: %s' % e)
    CondaRuntimeError: Runtime error: RuntimeError: Invalid mode: <conda.install.FileMode object at 0x7f3869e0a910>
```
I just tried this again, in the meantime I did no development nor fiddled with python at all and here is the output of a conda update command:

`conda update conda`

`bash: /home/maria/miniconda2/bin/conda: /opt/anaconda1anaconda2anaconda3/bin/python: bad interpreter: No such file or directory`

If I try to activate any env this is not working and I get the same message as conda update conda. 
I really haven't done anything to change my environment today so this is very odd.

**Comment by tyler-thetyrant** [comment]
@ciacicode I think your version of conda is too old to update itself to the recent versions.
I would uninstall and reinstall the newest [miniconda](https://conda.io/miniconda.html).

**Comment by ciacicode** [comment]
Just installed the new one but now all my environment folders are messed up...

**Comment by flovv** [comment]
Same here, calling conda from within R:

> install_tensorflow()
Creating r-tensorflow conda environment for TensorFlow installation...
Fetching package metadata ...An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : win-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : not installed
         python version : 3.6.0.final.0
       requests version : 2.12.4
       root environment : C:\PROGRA~3\ANACON~1  (writable)
    default environment : C:\PROGRA~3\ANACON~1
       envs directories : C:\PROGRA~3\ANACON~1\envs
          package cache : C:\PROGRA~3\ANACON~1\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : None
           offline mode : False
             user-agent : conda/4.3.8 requests/2.12.4 CPython/3.6.0 Windows/7 Windows/6.1.7601



`$ C:\PROGRA~3\ANACON~1\Scripts\conda-script.py create --yes --name r-tensorflow python=3.6`




    Traceback (most recent call last):
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\exceptions.py", line 617, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\cli\main.py", line 137, in _main
        exit_code = args.func(args, p)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\cli\main_create.py", line 68, in execute
        install(args, parser, 'create')
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\cli\install.py", line 210, in install
        unknown=index_args['unknown'], prefix=prefix)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 120, in get_index
        index = fetch_index(channel_priority_map, use_cache=use_cache)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 445, in fetch_index
        repodatas = _collect_repodatas(use_cache, urls)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 433, in _collect_repodatas
        repodatas = _collect_repodatas_serial(use_cache, urls)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 401, in _collect_repodatas_serial
        for url in urls]
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 401, in <listcomp>
        for url in urls]
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 141, in func
        res = f(*args, **kwargs)
      File "C:\PROGRA~3\ANACON~1\lib\site-packages\conda\core\index.py", line 391, in fetch_repodata
        with open(cache_path, 'w') as fo:
    PermissionError: [Errno 13] Permission denied: 'C:\\PROGRA~3\\ANACON~1\\pkgs\\cache\\2116b818.json'

Error: Error 1 occurred creating conda environment r-tensorflow
In addition: Warning message:
running command '"C:\PROGRA~3\ANACON~1\Scripts\conda.exe" "create" "--yes" "--name" "r-tensorflow" "python=3.6"' had status 1

**Comment by omegel** [comment]
Almost the same error that @lavi937 , so I first uninstalled all python versions and python-related program. I then reinstalled Anaconda3 (version 4.4.0), and when opening Anaconda Prompt, I get the following message:

> An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : win-64
          conda version : 4.3.21
       conda is private : False
      conda-env version : 4.3.21
    conda-build version : not installed
         python version : 3.6.1.final.0
       requests version : 2.14.2
       root environment : C:\Users\omege\Anaconda3  (writable)
    default environment : C:\Users\omege\Anaconda3
       envs directories : C:\Users\omege\Anaconda3\envs
                          C:\Users\omege\AppData\Local\conda\conda\envs
                          C:\Users\omege\.conda\envs
          package cache : C:\Users\omege\Anaconda3\pkgs
                          C:\Users\omege\AppData\Local\conda\conda\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : None
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.21 requests/2.14.2 CPython/3.6.1 Windows/10 Windows/10.0.15063
          administrator : False

`$ C:\Users\omege\Anaconda3\Scripts\conda-script.py ..checkenv cmd.exe C:\Users\omege\Anaconda3`




    Traceback (most recent call last):
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\cli\main.py", line 167, in main
        import conda.cli.activate as activate
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\cli\activate.py", line 12, in <module>
        from ..utils import shells
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\utils.py", line 13, in <module>
        from .gateways.disk.read import compute_md5sum
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\gateways\disk\read.py", line 22, in <module>
        from ...models.channel import Channel
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\models\channel.py", line 9, in <module>
        from ..base.context import context
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\base\context.py", line 18, in <module>
        from .._vendor.auxlib.path import expand
      File "C:\Users\omege\Anaconda3\lib\site-packages\conda\_vendor\auxlib\path.py", line 8, in <module>
        import pkg_resources
      File "<frozen importlib._bootstrap>", line 961, in _find_and_load
      File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 646, in _load_unlocked
      File "<frozen importlib._bootstrap>", line 616, in _load_backward_compatible
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2985, in <module>
        @_call_aside
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2971, in _call_aside
        f(*args, **kwargs)
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 3013, in _initialize_master_working_set
        dist.activate(replace=False)
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2544, in activate
        declare_namespace(pkg)
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2118, in declare_namespace
        _handle_ns(packageName, path_item)
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2058, in _handle_ns
        _rebuild_mod_path(path, packageName, module)
      File "C:\Users\omege\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.egg\pkg_resources\__init__.py", line 2087, in _rebuild_mod_path
        orig_path.sort(key=position_in_sys_path)
    AttributeError: '_NamespacePath' object has no attribute 'sort'

Note also that Anaconda Navigator won't open.

Any idea what is happening?

**Comment by omegel** [comment]
I just tried with the previous version (4.3.1), and I got the same issue (the Traceback is similar, except for the first few lines).

**Comment by omegel** [comment]
@lavi937: I managed to solve the problem on my side: It turned out I hadn't properly cleaned-up everything from a previous Python installation: there were files left in "C:\Users\me\AppData\Roaming\Python". I deleted the whole Python folder (also check for folders directly in C:\Users\me whose names start with a "." and delete the python-related ones. Probably some of these files where confusing Anaconda, in a way that I do not understand. 

After reinstalling Anaconda, everything seems to be working!

**Comment by kalefranz** [comment]
AttributeError: '_NamespacePath' object has no attribute 'sort'

Is a duplicate of #5483 and #5500. Fixed with #5496 in conda 4.3.22.

**Comment by kalefranz** [comment]
NameError: name 'compose' is not defined

Means you have an old/incompatible version of cytoolz in your environment.  Delete the `C:\Users\User\Anaconda3\lib\site-packages\cytoolz` directory and then `conda update cytoolz`.

**Comment by kalefranz** [comment]
CondaRuntimeError: Runtime error: RuntimeError: Invalid mode: <conda.install.FileMode object at 0x7fba0b2aff50>

Is an error that actually masks a real underlying error.  The masking has for sure been fixed.  Current version of conda is 4.3.24.

**Comment by kalefranz** [comment]
PermissionError: [Errno 13] Permission denied:

Fixed in 4.3.13.  Current version of conda is 4.3.24.

**Comment by kalefranz** [comment]
I don't see any stack traces here that are new and not fixed already.  Labeling ticket as duplicate and closing.
